### PR TITLE
Drop non-generic comparisons from Haskell AST

### PR DESCRIPTION
**Pull request by cocreature**
deleted code is the best code.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs [hunk 1]
```diff
@@ -534,6 +496,48 @@ decodeBuiltinFunction = \case
   LF1.BuiltinFunctionEQUAL_DECIMAL -> unsupportedDecimal
   LF1.BuiltinFunctionTEXT_TO_DECIMAL -> unsupportedDecimal
 
+  LF1.BuiltinFunctionEQUAL_INT64 -> unsupportedEqual
```
Comment (akrmn) [inline compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs:499]: Wouldn't this be a problem when reading DARs built with older versions of the compiler?

Comment (cocreature) [reply compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs:499]: Oh actually you’re right. Somehow I thought LF 1.8 already had those but it doesn’t. Turning it into a draft for now.

Comment (akrmn) [reply compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs:499]: ah! I was expecting a "that's taken care of elsewhere", glad I mentioned it. Please let me know how you work around it :)

Comment (cocreature) [reply compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs:499]: luckily the tests I was expecting to fail due to this did also fail. We technically only need to decode types for old LF versions. Maybe that works. But will probably prioritize doing some other stuff first and come back to this later.

**Comment by cocreature** [comment]
Closing for now

**Pull request closed without merging**